packages/timeline-state-resolver/src/integrations/tricaster/triCasterStateDiffer.ts

```diff
@@ -300,21 +301,29 @@ export class TriCasterStateDiffer {
 		source: CommandName.CROSSPOINT_SOURCE,
 	}
 
-	private effectCommandGenerator: CommandGeneratorFunction<
+	private keyerEffectCommandGenerator: CommandGeneratorFunction<
 		TriCasterTransitionEffect,
-		RequiredDeep<TriCasterMixEffectState | TriCasterKeyerState>
-	> = ({ value, target, state }) => {
-		const commands: TriCasterCommand[] = []
-		if (value === 'cut') {
-			return commands
-		}
-		if (typeof value === 'number') {
-			commands.push({ name: CommandName.SELECT_INDEX, target, value })
-		} else if (value === 'fade') {
-			commands.push({ name: CommandName.SELECT_INDEX, target, value: 1 })
+		RequiredDeep<TriCasterKeyerState>
+	> = this.effectCommandGenerator(CommandName.SELECT_INDEX)
+
+	private mixEffectEffectCommandGenerator: CommandGeneratorFunction<
+		TriCasterTransitionEffect,
+		RequiredDeep<TriCasterMixEffectState>
+	> = this.effectCommandGenerator(CommandName.SET_MIX_EFFECT_BIN_INDEX)
+
+	private effectCommandGenerator(
+		selectCommand: TriCasterGenericCommandName<number>
+	): CommandGeneratorFunction<TriCasterTransitionEffect, RequiredDeep<TriCasterKeyerState | TriCasterMixEffectState>> {
+		return ({ value, target, state }) => {
+			if (value === 'cut') {
```

Comment:
Should we perhaps have an enum rather than just using strings like `cut` to compare with?

Reply:
Yes, I can see the benefit of having a numeric enum here. Fade would no longer require extra code paths. I'll do that in the next PR, however